Index: src/modules/DashBoard/screens/DashBoard.js
===================================================================
diff --git a/src/modules/DashBoard/screens/DashBoard.js b/src/modules/DashBoard/screens/DashBoard.js
deleted file mode 100644
--- a/src/modules/DashBoard/screens/DashBoard.js	
+++ /dev/null	
@@ -1,9 +0,0 @@
-import React from 'react'
-
-function DashBoard() {
-    return (
-        <div>DashBoard</div>
-    )
-}
-
-export default DashBoard
\ No newline at end of file
Index: src/modules/DashBoard/index.js
===================================================================
diff --git a/src/modules/DashBoard/index.js b/src/modules/DashBoard/index.js
deleted file mode 100644
--- a/src/modules/DashBoard/index.js	
+++ /dev/null	
@@ -1,1 +0,0 @@
-export { default as DashBoard } from './screens/DashBoard'
\ No newline at end of file
Index: src/layout/DefaultLayout/Header/utils/index.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/layout/DefaultLayout/Header/utils/index.js b/src/layout/DefaultLayout/Header/utils/index.js
new file mode 100644
--- /dev/null	
+++ b/src/layout/DefaultLayout/Header/utils/index.js	
@@ -0,0 +1,13 @@
+// import { login } from "../api"
+// import { useEffect, useState } from 'react'
+// export const useLogin = (params) => {
+//       const [data, setData] = useState([])
+//       async function fetchData() {
+//             const response = await login({ status: 1, ...params })
+//             if (response.data.status) setData(response.data.data)
+//       }
+//       useEffect(() => {
+//             fetchData()
+//       }, [JSON.stringify(params)])
+//       return data
+// }
\ No newline at end of file
Index: src/layout/index.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>export { default as DefaultLayout } from \"./DefaultLayout\"
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/layout/index.js b/src/layout/index.js
--- a/src/layout/index.js	
+++ b/src/layout/index.js	
@@ -1,1 +1,4 @@
-export { default as DefaultLayout } from "./DefaultLayout"
\ No newline at end of file
+export { default as DefaultLayout } from "./DefaultLayout"
+export { default as AdminLayout } from "./AdminLayout/AdminLayout"
+
+
Index: src/layout/style/index.css
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/layout/style/index.css b/src/layout/style/index.css
new file mode 100644
--- /dev/null	
+++ b/src/layout/style/index.css	
@@ -0,0 +1,108 @@
+.header_bar {
+      background-color: #1e1d1d;
+  
+}
+.sidebar {
+    height: 100vh;
+    background-color: #1e1d1d;
+    border-right: 1px #7C7272 solid ;
+    position: absolute;
+    left: -15vw;
+    height: 100%;
+    width: 15vw;
+    transition: left 0.3s ease-in-out;
+    border-bottom-right-radius: 15px;
+  }
+  .sidebar.open {
+    left: 0;
+  }
+  .content.open {
+    left: 15vw;
+  }
+  .content {
+    transition: left 0.3s ease-in-out;
+  }
+  .content.open {
+    left: 15vw; 
+  }
+  .header.open, .content.open {
+    margin-left: 15vw; 
+}
+
+.header {
+    transition: margin-left 0.3s ease-in-out;
+}
+
+.content {
+    transition: margin-left 0.3s ease-in-out;
+}
+.text_hover:hover{
+    background-image: linear-gradient(to right, #029D43 ,#1e1d1d);
+    border-left: 5px  #30F274 solid;
+}
+.p-card {
+    background: #000000 !important;
+    color: #FFFFFF;
+    box-shadow: 0 2px 1px -1px rgba(0, 0, 0, 0.2), 0 1px 1px 0 rgba(0, 0, 0, 0.14), 0 1px 3px 0 rgba(0, 0, 0, 0.12);
+    border-radius: 6px;
+}
+.p-card-subtitle{
+    color: #656565;
+}
+.container {
+    position: relative;
+
+  }
+.image_overlay {
+    display: block;
+    width: 100%;
+    height: auto;
+}
+  
+.container {
+    position: relative;
+}
+
+.image_overlay {
+    display: block;
+    width: 100%;
+    height: auto;
+}
+
+.overlay {
+    position: absolute;
+    top: 0;
+    bottom: 0;
+    left: 0;
+    right: 0;
+    height: 100%;
+    width: 100%;
+    opacity: 0;
+    transition: .5s ease;
+    background-color: #30F274;
+}
+
+.play {
+    display: none;
+    
+}
+
+.container:hover .overlay {
+    opacity: 0.3;
+}
+
+.container:hover .play {
+    display: block;
+    /* Có thể thêm thuộc tính opacity ở đây nếu muốn */
+}
+
+.child {
+    color: white;
+    font-size: 20px;
+    position: absolute;
+    top: 50%;
+    left: 50%;
+    transform: translate(-50%, -50%);
+    text-align: center;
+}
+
Index: src/index.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>import React from 'react';\r\nimport ReactDOM from 'react-dom/client';\r\nimport App from '@/App';\r\nimport reportWebVitals from './reportWebVitals';\r\nimport \"primereact/resources/themes/lara-light-cyan/theme.css\";\r\nimport 'primeicons/primeicons.css';\r\nimport 'primeflex/primeflex.css'\r\nimport \"./index.css\"\r\n\r\nconst root = ReactDOM.createRoot(document.getElementById('root'));\r\nroot.render(\r\n  <React.StrictMode>\r\n    <App />\r\n  </React.StrictMode>\r\n);\r\n\r\n// If you want to start measuring performance in your app, pass a function\r\n// to log results (for example: reportWebVitals(console.log))\r\n// or send to an analytics endpoint. Learn more: https://bit.ly/CRA-vitals\r\nreportWebVitals();\r\n
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/index.js b/src/index.js
--- a/src/index.js	
+++ b/src/index.js	
@@ -1,20 +1,18 @@
 import React from 'react';
 import ReactDOM from 'react-dom/client';
 import App from '@/App';
-import reportWebVitals from './reportWebVitals';
+
 import "primereact/resources/themes/lara-light-cyan/theme.css";
-import 'primeicons/primeicons.css';
+import 'primeicons/primeicons.css' // icons
 import 'primeflex/primeflex.css'
-import "./index.css"
-
-const root = ReactDOM.createRoot(document.getElementById('root'));
+import "./App.css"
+import { Provider } from 'react-redux';
+import store from './redux/store';
+const root = ReactDOM.createRoot(document.getElementById('root'))
 root.render(
-  <React.StrictMode>
+  <Provider store={store}>
     <App />
-  </React.StrictMode>
+  </Provider>
 );
 
-// If you want to start measuring performance in your app, pass a function
-// to log results (for example: reportWebVitals(console.log))
-// or send to an analytics endpoint. Learn more: https://bit.ly/CRA-vitals
-reportWebVitals();
+
Index: src/lib/request.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/lib/request.js b/src/lib/request.js
new file mode 100644
--- /dev/null	
+++ b/src/lib/request.js	
@@ -0,0 +1,9 @@
+import { clientApi } from '../axios/index'
+
+export const postData = (url, params, timeout = 600000) => {
+      return clientApi.post(url, params, { timeout })
+}
+
+export const getData = (url, params, timeout = 600000) => {
+      return clientApi.get(url, { params, timeout })
+}
\ No newline at end of file
Index: src/modules/Home/api/index.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/modules/Home/api/index.js b/src/modules/Home/api/index.js
new file mode 100644
--- /dev/null	
+++ b/src/modules/Home/api/index.js	
@@ -0,0 +1,1 @@
+import { postData, getData } from "@/lib/request";
Index: src/routes/index.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>import { DashBoard } from \"@/modules/DashBoard\"\r\n\r\n\r\nconst publicRoutes = [\r\n    { path: '/', component: DashBoard, layout: true }\r\n]\r\nconst privateRoutes = []\r\n\r\nexport { privateRoutes, publicRoutes } 
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/routes/index.js b/src/routes/index.js
--- a/src/routes/index.js	
+++ b/src/routes/index.js	
@@ -1,9 +1,20 @@
-import { DashBoard } from "@/modules/DashBoard"
+
+import { AdminLayout } from "@/layout"
+import { Home } from "@/modules/Home"
+import { HomeAdmin } from "@/modules/HomeAdmin"
+import { ListSong } from "@/modules/ListSong"
+import { ListUser } from "@/modules/ListUser"
 
 
 const publicRoutes = [
-    { path: '/', component: DashBoard, layout: true }
+    { path: '/', component: Home, layout: 'user' },
 ]
-const privateRoutes = []
+const privateRoutes = [
+
+    { path: '/admin', component: HomeAdmin, layout: 'admin' },
+    { path: '/list_user', component: ListUser, layout: 'admin' },
+    { path: '/list_song', component: ListSong, layout: 'admin' },
+
+]
 
 export { privateRoutes, publicRoutes } 
\ No newline at end of file
Index: src/modules/Home/screens/Banner.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/modules/Home/screens/Banner.js b/src/modules/Home/screens/Banner.js
new file mode 100644
--- /dev/null	
+++ b/src/modules/Home/screens/Banner.js	
@@ -0,0 +1,25 @@
+import { test } from '@/axios/test';
+import Cardz from '@/components/Card';
+import { Carousel } from 'primereact/carousel';
+import React, { useEffect, useState } from 'react'
+function Banner() {
+      const itemTemplate = (item) => {
+            return (
+                  <img src="https://picsum.photos/200/300" alt={item.alt} style={{ width: '100%', height: '200px' }} />
+            );
+      }
+      return (
+
+            <div >
+
+
+
+
+                  <Carousel value={test.map(d => d.url)} itemTemplate={itemTemplate}></Carousel>
+
+            </div>
+
+      )
+}
+
+export default Banner
Index: src/modules/Home/screens/Home.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/modules/Home/screens/Home.js b/src/modules/Home/screens/Home.js
new file mode 100644
--- /dev/null	
+++ b/src/modules/Home/screens/Home.js	
@@ -0,0 +1,81 @@
+import { clientApi } from '@/axios'
+import { test } from '@/axios/test'
+import Cardz from '@/components/Card'
+import Card from '@/components/Card'
+import Title from '@/components/Title'
+import axios from 'axios'
+
+import React, { useEffect, useState } from 'react'
+import Banner from './Banner'
+import TrendRegular from './TrendRegular'
+
+
+import ads from '@/layout/img/ads.png'
+import NewRelease from './NewRelease'
+import Images from './Images'
+import Partner from './Partner'
+
+function Home() {
+    return (
+        <>
+
+            <div className="grid">
+                <div className="col-9">
+                    <Banner />
+                    <Title title="Mùa chay" icon='pi-angle-double-right' subTitle="Xem thêm" />
+                    <div className="grid">
+                        {
+                            test.slice(0, 8).map(d =>
+                                <div className="col-3">
+                                    <Cardz song="Cảm mến ân tình" src="https://picsum.photos/200/300" sing="Thế Quyền" />
+                                </div>
+                            )
+                        }
+                    </div >
+                </div>
+                <div className="col-3">
+                    <TrendRegular />
+                </div>
+            </div>
+            <div className="grid">
+                <div className="col-9">
+                    <Title title="Mùa chay" icon='pi-angle-double-right' subTitle="Xem thêm" />
+                    <div className="grid">
+                        {
+                            test.slice(0, 8).map(d =>
+                                <div className="col-3">
+                                    <Cardz song="Cảm mến ân tình" src="https://picsum.photos/200/300" sing="Thế Quyền" />
+                                </div>
+                            )
+                        }
+                    </div >
+                </div>
+                <div className="col-3 relative">
+                    <img src={ads} alt="" className='absolute right-0' />
+                </div>
+            </div>
+            <NewRelease />
+            <Images />
+            <div className="grid">
+                <div className="col-9">
+                    <Title title="Bài đọc" icon='pi-angle-double-right' subTitle="Xem thêm" />
+                    <div className="grid">
+                        {
+                            test.slice(0, 8).map(d =>
+                                <div className="col-3">
+                                    <Cardz song="Cảm mến ân tình" src="https://picsum.photos/200/300" sing="Thế Quyền" />
+                                </div>
+                            )
+                        }
+                    </div >
+                </div>
+                <div className="col-3 ">
+
+                </div>
+            </div>
+
+        </>
+    )
+}
+
+export default Home
\ No newline at end of file
Index: src/modules/Home/screens/Images.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/modules/Home/screens/Images.js b/src/modules/Home/screens/Images.js
new file mode 100644
--- /dev/null	
+++ b/src/modules/Home/screens/Images.js	
@@ -0,0 +1,18 @@
+import { test } from '@/axios/test'
+import React from 'react'
+
+function Images() {
+      return (
+            <div className='grid py-4'>
+                  {
+                        test.slice(0, 3).map(d =>
+                              <div className="col-4">
+                                    <img src="https://picsum.photos/200/300" className='w-full h-18rem border-round-sm' alt="" />
+                              </div>
+                        )
+                  }
+            </div>
+      )
+}
+
+export default Images
Index: src/modules/Home/screens/NewRelease.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/modules/Home/screens/NewRelease.js b/src/modules/Home/screens/NewRelease.js
new file mode 100644
--- /dev/null	
+++ b/src/modules/Home/screens/NewRelease.js	
@@ -0,0 +1,44 @@
+import { test } from '@/axios/test'
+import Title from '@/components/Title'
+import React from 'react'
+import { Link } from 'react-router-dom'
+
+function NewRelease() {
+      return (
+            <div>
+                  <div className="grid">
+                        <div className="col-9">
+                              <Title title="Mới phát hành" icon='pi-angle-double-right' subTitle="Xem thêm" />
+                              <div>
+                                    {
+                                          test.slice(0, 3).map(d =>
+                                                <div className='grid justify-content-center align-items-center text-center'>
+                                                      <div className="col-1">
+                                                            <img src="https://picsum.photos/200/300" width="100px" height='100px' alt="" />
+                                                      </div>
+                                                      <div className="col-5">
+                                                            <Link to="" className=' no-underline text-xl' style={{ color: '#03CE58' }} > Trong tình yêu của mẹ</Link>
+                                                      </div>
+                                                      <div className="col-2">
+                                                            <Link to="" className='text-white no-underline text-xl' > Thế Quyền</Link>
+                                                      </div>
+                                                      <div className="col-2">
+                                                            <span style={{ color: '#656565' }} >04:25</span>
+                                                      </div>
+                                                      <div className="col-2">
+                                                            <i className="pi pi-plus text-lg" style={{ color: "#E5AC1A" }}  ></i>
+                                                      </div>
+                                                </div>
+                                          )
+                                    }
+                              </div>
+                        </div>
+                        <div className="col-3">
+
+                        </div>
+                  </div>
+            </div>
+      )
+}
+
+export default NewRelease
Index: src/modules/Home/screens/Partner.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/modules/Home/screens/Partner.js b/src/modules/Home/screens/Partner.js
new file mode 100644
--- /dev/null	
+++ b/src/modules/Home/screens/Partner.js	
@@ -0,0 +1,21 @@
+import { test } from '@/axios/test'
+import React from 'react'
+
+function Partner() {
+      return (
+            <div className='partner mr-2 px-2'>
+                  <h1 className='text-center m-0 p-4' style={{ color: '#03CE58' }}>Các đối tác âm nhạc</h1>
+                  <div className="grid">
+                        {
+                              test.slice(0, 12).map(d =>
+                                    <div className="col-2">
+                                          <img src="https://picsum.photos/200/300" className='w-full h-8rem border-round-sm' alt="" />
+                                    </div>
+                              )
+                        }
+                  </div>
+            </div>
+      )
+}
+
+export default Partner
Index: src/layout/AdminLayout/SideBar/SideBarAdmin.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/layout/AdminLayout/SideBar/SideBarAdmin.js b/src/layout/AdminLayout/SideBar/SideBarAdmin.js
new file mode 100644
--- /dev/null	
+++ b/src/layout/AdminLayout/SideBar/SideBarAdmin.js	
@@ -0,0 +1,36 @@
+import React from 'react'
+
+import { Link } from "react-router-dom";
+import '../../style/index.css'
+
+function SideBarAdmin() {
+      return (
+            <div className='card  sidebar_admin '>
+                  <ul className='m-0 p-0'>
+                        <li className='p-3 flex gap-3  text-xl text-white'>
+                              <i className="pi pi-home text-xl" ></i>
+                              <Link className='no-underline text-white'>Trang chủ</Link>
+                        </li>
+                        <li className='p-3 flex gap-3  text-xl text-white'>
+                              <i className="pi pi-users text-xl" ></i>
+                              <Link to="/list_user" className='no-underline text-white'>Danh sách người dùng</Link>
+                        </li>
+                        <li className='p-3 flex gap-3  text-xl text-white'>
+                              <i className="pi pi-list text-xl" ></i>
+                              <Link to="/list_song" className='no-underline text-white'>Danh sách bài hát</Link>
+                        </li>
+                        <li className='p-3 flex gap-3  text-xl text-white'>
+                              <i className="pi pi-book text-xl" ></i>
+                              <Link className='no-underline text-white'>Bài đọc</Link>
+                        </li>
+                        <li className='p-3 flex gap-3  text-xl text-white'>
+                              <i className="pi pi-cog pi pi-spin text-xl" ></i>
+                              <Link className='no-underline text-white'>Chỉnh sửa tổng quan</Link>
+                        </li>
+                  </ul>
+            </div>
+      )
+
+
+}
+export default SideBarAdmin
Index: src/layout/AdminLayout/AdminLayout.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/layout/AdminLayout/AdminLayout.js b/src/layout/AdminLayout/AdminLayout.js
new file mode 100644
--- /dev/null	
+++ b/src/layout/AdminLayout/AdminLayout.js	
@@ -0,0 +1,23 @@
+import React from 'react'
+import SideBarAdmin from './SideBar/SideBarAdmin'
+import '../style/index.css'
+import HeaderAdmin from './Header/HeaderAdmin';
+
+function AdminLayout({ children }) {
+      return (
+            <div className="">
+                  <HeaderAdmin />
+                  <div className="grid p-2">
+                        <div className="col-2 pl-0 ">
+                              <SideBarAdmin />
+                        </div>
+                        <div className="col-10">
+                              {children}
+                        </div>
+                  </div>
+
+            </div >
+      )
+}
+
+export default AdminLayout
Index: src/layout/DefaultLayout/AudioPlay/screens/AudioPlay.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/layout/DefaultLayout/AudioPlay/screens/AudioPlay.js b/src/layout/DefaultLayout/AudioPlay/screens/AudioPlay.js
new file mode 100644
--- /dev/null	
+++ b/src/layout/DefaultLayout/AudioPlay/screens/AudioPlay.js	
@@ -0,0 +1,106 @@
+import React, { useEffect, useRef, useState } from 'react';
+import { Button } from 'primereact/button';
+import { ProgressBar } from 'primereact/progressbar';
+import { Link } from 'react-router-dom';
+
+const AudioPlay = () => {
+      const audioRef = useRef();
+      const progressBarContainerRef = useRef();
+      const [isPlaying, setIsPlaying] = useState(false);
+      const [currentTime, setCurrentTime] = useState(0);
+      const [duration, setDuration] = useState(0);
+      const [progress, setProgress] = useState(0);
+      const toggleAudio = () => {
+            if (isPlaying) {
+                  audioRef.current.pause();
+            } else {
+                  audioRef.current.play();
+            }
+            setIsPlaying(!isPlaying);
+      }
+      useEffect(() => {
+            const audio = audioRef.current;
+            const updateProgress = () => {
+                  setProgress((audio.currentTime / audio.duration) * 100);
+                  setCurrentTime(audio.currentTime);
+            };
+            const setAudioDuration = () => {
+                  setDuration(audio.duration);
+            };
+            audio.addEventListener('timeupdate', updateProgress);
+            audio.addEventListener('loadedmetadata', setAudioDuration);
+
+            return () => {
+                  audio.removeEventListener('timeupdate', updateProgress);
+                  audio.removeEventListener('loadedmetadata', setAudioDuration);
+            };
+      }, []);
+      const formatTime = (timeInSeconds) => {
+            const minutes = Math.floor(timeInSeconds / 60);
+            const seconds = Math.floor(timeInSeconds % 60);
+            return `${minutes}:${seconds < 10 ? '0' : ''}${seconds}`;
+      };
+      const handleProgressBarClick = (e) => {
+            const rect = progressBarContainerRef.current.getBoundingClientRect();
+            const x = e.clientX - rect.left;
+            const clickedValue = x * audioRef.current.duration / rect.width;
+            audioRef.current.currentTime = clickedValue;
+      };
+      const [isClicked, setIsClicked] = useState(false);
+
+      const handleClick = () => {
+            setIsClicked(!isClicked);
+      }
+      return (
+            <div style={{ position: 'fixed', bottom: 0, background: '#989898' }} className='w-full h-6rem' >
+                  <audio ref={audioRef}>
+                        <source src="https://firebasestorage.googleapis.com/v0/b/nodejs-9c5e8.appspot.com/o/songs%2F1Phut-Andiez-7632303.mp3?alt=media&token=4a162195-a4f6-4fa4-a095-12ec3561f0f1" type="audio/mpeg" />
+                        Trình duyệt của bạn không hỗ trợ thẻ audio.
+                  </audio>
+                  <div className="grid align-items-center">
+                        <div className="col-3">
+                              <ul className='p-0 m-0 flex  '>
+                                    <div className='flex  gap-2 flex-column '>
+                                          <li className='flex align-items-center px-3'>
+                                                <img src="https://picsum.photos/200/300" alt="" width="80px" height="80px" className='border-round-lg' />
+                                                <div className='flex flex-column justify-content-center p-3 gap-2'>
+                                                      <Link to="" className='text-white no-underline text-xl'>Con đường Chúa đã đi</Link>
+                                                      <Link to="" className='text-white no-underline'>Thế Quyền</Link>
+                                                </div>
+                                          </li>
+                                    </div>
+                              </ul>
+                        </div>
+                        <div className="col-2 ">
+                              <div className='flex gap-3' >
+                                    <Button
+                                          icon="pi pi-heart"
+                                          className={isClicked ? '' : 'p-button-outlined'}
+                                          style={isClicked ? { background: '#03CE58', border: 'none' } : {}}
+                                          onClick={handleClick}
+                                    />
+                                    <Button style={{ background: '#03CE58', border: 'none' }} icon="pi pi-step-backward-alt" />
+                                    <Button style={{ background: '#03CE58', border: 'none' }} icon={isPlaying ? "pi pi-pause" : "pi pi-play"} onClick={toggleAudio} />
+                                    <Button style={{ background: '#03CE58', border: 'none' }} icon="pi pi-step-forward-alt" />
+                              </div>
+                        </div>
+                        <div className="col-5">
+                              <div className="flex justify-content-center  gap-3"   >
+                                    <div> {formatTime(currentTime)}</div>
+                                    <div ref={progressBarContainerRef} onClick={handleProgressBarClick}>
+                                          <ProgressBar value={progress} className='w-30rem h-1rem' showValue={false} />
+                                    </div>
+                                    <div> {formatTime(duration - currentTime)}</div>
+                              </div>
+                        </div>
+                        <div className="col-2 flex gap-3">
+                              <Button style={{ background: '#03CE58', border: 'none' }} icon="pi pi-question-circle" />
+                              <Button style={{ background: '#03CE58', border: 'none' }} icon="pi pi-sync" />
+                              <Button style={{ background: '#03CE58', border: 'none' }} icon="pi pi-list" />
+                        </div>
+                  </div>
+            </div>
+      );
+};
+
+export default AudioPlay;
Index: src/layout/DefaultLayout/footer/screens/Footer.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/layout/DefaultLayout/footer/screens/Footer.js b/src/layout/DefaultLayout/footer/screens/Footer.js
new file mode 100644
--- /dev/null	
+++ b/src/layout/DefaultLayout/footer/screens/Footer.js	
@@ -0,0 +1,68 @@
+import React from 'react'
+import logo from '@/layout/img/logo.png'
+import { Link } from 'react-router-dom'
+function Footer() {
+      return (
+            <div className='footer grid'>
+                  <div className="col-6 pl-8">
+                        <div className='flex'>
+                              <img width="88px" src={logo} alt="" />
+                              <h1>VIETTHANHCA</h1>
+                        </div>
+                        <div className=''>
+                              <p style={{ maxWidth: '370px' }}>Vietthanhca là nguồn nhạc miễn phí bản quyền. Nâng cao dự án của bạn với bộ sưu tập đa dạng của chúng tôi. Đơn giản hóa quy trình cấp phép và giải phóng khả năng sáng tạo của bạn. Tham gia với chúng tôi ngay hôm nay!</p>
+                        </div>
+
+                  </div>
+                  <div className="col-2">
+                        <h3 style={{ color: '#03CE58' }}>Sản phẩm</h3>
+                        <ul className='list-none p-0 flex flex-column gap-3'>
+                              <li>
+                                    <Link className='text-white'>Thịnh hành</Link>
+                              </li>
+                              <li>
+                                    <Link className='text-white'>Instrument Music</Link>
+                              </li>
+                              <li>
+                                    <Link className='text-white'>Thể loại </Link>
+                              </li>
+                              <li>
+                                    <Link className='text-white'>Licensing </Link>
+                              </li>
+                        </ul>
+                  </div>
+                  <div className="col-2">
+                        <h3 style={{ color: '#03CE58' }}>Hỗ trợ</h3>
+                        <ul className='list-none p-0 flex flex-column gap-3'>
+                              <li>
+                                    <Link className='text-white'>FAQ</Link>
+                              </li>
+                              <li>
+                                    <Link className='text-white'>Liên hệ </Link>
+                              </li>
+                              <li>
+                                    <Link className='text-white'>Chính sách bảo mật</Link>
+                              </li>
+                              <li>
+                                    <Link className='text-white'>Điều khoản dịch vụ</Link>
+                              </li>
+                        </ul>
+                  </div>
+                  <div className="col-2">
+                        <h3 style={{ color: '#03CE58' }}>Hỗ trợ</h3>
+                        <p className='text-xl'>
+                              Để biết các cập nhật và tin tức gần đây, hãy theo dõi nguồn cấp dữ liệu truyền thông xã hội của chúng tôi
+                        </p>
+                        <div className="flex gap-3">
+                              <i className="pi pi-youtube" style={{ fontSize: '2rem', color: '#03CE58' }}></i>
+                              <i className="pi pi-instagram" style={{ fontSize: '2rem', color: '#03CE58' }}></i>
+                              <i className="pi pi-facebook" style={{ fontSize: '2rem', color: '#03CE58' }}></i>
+                              <i className="pi pi-twitter" style={{ fontSize: '2rem', color: '#03CE58' }}></i>
+                              <i className="pi pi-envelope" style={{ fontSize: '2rem', color: '#03CE58' }}></i>
+                        </div>
+                  </div>
+            </div>
+      )
+}
+
+export default Footer
Index: src/layout/DefaultLayout/Header/api/index.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/layout/DefaultLayout/Header/api/index.js b/src/layout/DefaultLayout/Header/api/index.js
new file mode 100644
--- /dev/null	
+++ b/src/layout/DefaultLayout/Header/api/index.js	
@@ -0,0 +1,2 @@
+
+
Index: src/layout/DefaultLayout/Header/screens/index.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/layout/DefaultLayout/Header/screens/index.js b/src/layout/DefaultLayout/Header/screens/index.js
new file mode 100644
--- /dev/null	
+++ b/src/layout/DefaultLayout/Header/screens/index.js	
@@ -0,0 +1,87 @@
+// Header.js
+import React, { useState, useRef } from 'react'
+import { InputText } from 'primereact/inputtext';
+import { Button } from 'primereact/button';
+import '../../../style/index.css'
+
+import Login from './Login';
+import SignUp from './SignUp';
+import { useNavigate } from "react-router-dom";
+import { useSelector, useDispatch } from 'react-redux';
+import { Menu } from 'primereact/menu';
+const Header = props => {
+    const [visibleSignUp, setVisibleSignUp] = useState(false);
+    const [visibleLogin, setVisibleLogin] = useState(false);
+    const navigate = useNavigate();
+    const menuRight = useRef(null);
+    const dispatch = useDispatch();
+    const isLoggedIn = useSelector((state) => state.user.isLoggedIn);
+    const userInfo = useSelector((state) => state.user.userInfo);
+    const items = [
+        {
+            label: 'Cá nhân',
+            items: [
+                {
+                    label: 'Chủ đề',
+                    icon: 'pi pi-palette'
+                },
+                {
+                    label: 'Tài khoản',
+                    icon: 'pi pi-user-edit'
+                },
+                {
+                    label: 'Tải về',
+                    icon: 'pi pi-download'
+                },
+                {
+                    label: 'Hỗ trợ',
+                    icon: 'pi pi-info-circle'
+                },
+                {
+                    label: 'Cài đặt',
+                    icon: 'pi pi-cog'
+                },
+                {
+                    label: 'Đăng xuất',
+                    icon: 'pi pi-sign-out'
+
+                }
+            ]
+        }
+    ];
+    return (
+        <div className={`header`}>
+            <SignUp visibleSignUp={visibleSignUp} setVisibleSignUp={setVisibleSignUp} />
+            <Login visibleLogin={visibleLogin} setVisibleLogin={setVisibleLogin} />
+            <div className="card h-5rem header_bar ">
+                <div className=" flex justify-content-between p-3">
+                    <Button onClick={props.onClick} icon="pi pi-align-justify" />
+                    <div className="w-6 flex gap-2 ">
+                        <InputText type="text" placeholder="Bạn muốn nghe gì ?" className='w-full' />
+                        <Button icon="pi pi-search" label='Tìm kiếm' className='text-black-alpha-80 bg-white w-3 border-none' />
+                    </div>
+                    {isLoggedIn ?
+                        <>
+                            {
+                                userInfo?.admin === true ?
+                                    <Button label='Đi đến trang quản trị' onClick={() => navigate('/admin')} /> : null
+                            }
+                            <div className="flex gap-3">
+                                <Menu model={items} popup ref={menuRight} id="popup_menu_right" popupAlignment="right" />
+                                <Button icon="pi pi-user" onClick={(event) => menuRight.current.toggle(event)} aria-controls="popup_menu_right" aria-haspopup />
+
+                            </div>
+                        </>
+                        :
+                        <div className="flex gap-3">
+                            <Button icon="pi pi-user-plus" label='Đăng ký' className='text-black-alpha-80 bg-white border-none' rounded onClick={() => setVisibleSignUp(true)} />
+                            <Button icon="pi pi-sign-in" label='Đăng nhập' className='text-black-alpha-80 bg-white border-none' rounded onClick={() => setVisibleLogin(true)} />
+                        </div>
+                    }
+                </div>
+            </div>
+        </div>)
+}
+
+
+export default Header;
Index: src/layout/DefaultLayout/Header/screens/Login.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/layout/DefaultLayout/Header/screens/Login.js b/src/layout/DefaultLayout/Header/screens/Login.js
new file mode 100644
--- /dev/null	
+++ b/src/layout/DefaultLayout/Header/screens/Login.js	
@@ -0,0 +1,71 @@
+// Login.js
+import React, { useState } from 'react';
+import { Dialog } from 'primereact/dialog';
+import InputForm from '@/components/Form';
+import { Checkbox } from 'primereact/checkbox';
+import { Button } from 'primereact/button';
+import axios from 'axios';
+import { clientApi } from '@/axios';
+import { useDispatch, useSelector } from 'react-redux';
+import { useNavigate } from 'react-router-dom';
+
+import { loginSuccess } from '@/redux/userSlice';
+
+
+
+function Login(props) {
+      const { visibleLogin, setVisibleLogin } = props;
+      const [checked, setChecked] = useState(false);
+      const dispatch = useDispatch();
+      const navigate = useNavigate();
+      const [infos, setInfos] = useState({
+            username: '',
+            password: '',
+      });
+      const handleData = async () => {
+            try {
+                  const response = await clientApi.post('/user/login', infos);
+                  const token = response.data.accessToken;
+                  const userInfo = response.data.data;
+                  dispatch(loginSuccess({ token, userInfo }));
+                  setVisibleLogin(false)
+                  localStorage.setItem('token', token);
+
+            } catch (error) {
+                  console.error(error);
+            }
+      };
+      const footerContent = (
+            <div>
+                  <Button label="Hủy" icon="pi pi-times" onClick={() => setVisibleLogin(false)} severity="danger" />
+                  <Button label="Đăng Nhập" onClick={handleData} severity="secondary" icon="pi pi-sign-in" />
+            </div>
+      );
+
+      return (
+            <div>
+                  <Dialog header="Đăng nhập" visible={visibleLogin} style={{ width: '50vw', top: '10px', position: 'fixed' }} onHide={() => setVisibleLogin(false)} footer={footerContent}>
+                        <InputForm
+                              className="w-full"
+                              value={infos.username}
+                              label="Tài Khoản"
+                              onChange={(e) => setInfos({
+                                    ...infos, username: e.target.value
+                              })} />
+
+                        <InputForm
+                              value={infos.password}
+                              onChange={(e) => setInfos({ ...infos, password: e.target.value })} label="Mật khẩu"
+                              className="w-full"
+                              type={checked ? 'text' : 'password'}
+                        />
+                        <div className="flex align-items-center px-2">
+                              <Checkbox value="Cheese" onChange={e => setChecked(e.checked)} checked={checked} />
+                              <label className="ml-2">Hiển thị mật khẩu</label>
+                        </div>
+                  </Dialog>
+            </div>
+      );
+}
+
+export default Login;
Index: src/layout/DefaultLayout/Header/screens/SignUp.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/layout/DefaultLayout/Header/screens/SignUp.js b/src/layout/DefaultLayout/Header/screens/SignUp.js
new file mode 100644
--- /dev/null	
+++ b/src/layout/DefaultLayout/Header/screens/SignUp.js	
@@ -0,0 +1,68 @@
+import React, { useState } from 'react'
+import { Dialog } from 'primereact/dialog';
+import InputForm from '@/components/Form';
+
+import { Button } from 'primereact/button';
+import { Checkbox } from 'primereact/checkbox';
+import { clientApi } from '@/axios';
+
+function SignUp(props) {
+      const { visibleSignUp, setVisibleSignUp, setVisibleLogin } = props
+      const [checked, setChecked] = useState(false);
+
+
+      const [infos, setInfos] = useState({
+            username: '',
+            password: '',
+            email: ''
+      })
+      const handleData = async () => {
+
+            try {
+                  await clientApi.post('/user/signup', infos);
+                  setVisibleSignUp(false)
+                  setVisibleLogin(true)
+            } catch (error) {
+                  console.error(error);
+
+            }
+      };
+      const footerContent = (
+            <div>
+                  <Button label="Hủy" icon="pi pi-times" onClick={() => setVisibleSignUp(false)} severity="danger" />
+                  <Button label="Xác nhận" onClick={handleData} severity="secondary" icon="pi pi-check-square" />
+            </div>
+      );
+      return (
+            <div>
+                  <Dialog header="Đăng ký" visible={visibleSignUp} style={{ width: '50vw', top: '10px', position: 'fixed' }} onHide={() => setVisibleSignUp(false)} footer={footerContent}>
+                        <InputForm
+                              value={infos.email}
+                              onChange={(e) => setInfos({ ...infos, email: e.target.value })} label="Email"
+                              className="w-full"
+                        />
+                        <InputForm
+                              className="w-full"
+                              value={infos.username}
+                              label="Tài Khoản"
+                              onChange={(e) => setInfos({
+                                    ...infos, username: e.target.value
+                              })} />
+
+                        <InputForm
+                              value={infos.password}
+                              onChange={(e) => setInfos({ ...infos, password: e.target.value })} label="Mật khẩu"
+                              className="w-full"
+
+                        />
+                        <div className="flex align-items-center px-2">
+                              <Checkbox value="Cheese" onChange={e => setChecked(e.checked)} checked={checked} />
+                              <label className="ml-2">Hiển thị mật khẩu</label>
+                        </div>
+                  </Dialog>
+
+            </div>
+      )
+}
+
+export default SignUp
\ No newline at end of file
Index: src/modules/ListUser/screens/ListUser.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/modules/ListUser/screens/ListUser.js b/src/modules/ListUser/screens/ListUser.js
new file mode 100644
--- /dev/null	
+++ b/src/modules/ListUser/screens/ListUser.js	
@@ -0,0 +1,55 @@
+import React from 'react'
+import { DataTable } from 'primereact/datatable';
+import { Column } from 'primereact/column';
+import { useListUser } from '../utils';
+import DataTablez from '@/components/Datatablez';
+import { Button } from 'primereact/button';
+import InputForm from '@/components/Form';
+const Header = () => {
+      return (
+            <>
+                  <div className='grid '>
+                        <div className="col-3">
+                              <InputForm label="Lọc theo người dùng" />
+                        </div>
+                        <div className="col-3">
+                              <InputForm label="Lọc theo email" />
+                        </div>
+                        <div className="col-3">
+                              <InputForm label="Lọc theo chức vụ" />
+                        </div>
+                  </div>
+            </>
+      )
+}
+function ListUser() {
+      const data = useListUser()
+
+      return (
+            <div className='element'>
+                  <div className="">
+                        <Header />
+                  </div>
+                  <div className="element">
+                        <Button label='Thêm mới' className='mb-3' />
+                        <DataTablez value={data} >
+                              <Column field="username" header="Tên người dùng" ></Column>
+                              <Column field="email" header="Email"></Column>
+                              <Column body={
+                                    (rowData) => {
+                                          return rowData.admin === true
+                                                ?
+                                                <span className='bg-cyan-400 p-3 border-round-xl'>Admin</span>
+                                                :
+                                                <span className='bg-yellow-300 p-3 border-round-xl '>Người dùng</span>
+                                    }
+                              } header="Chức vụ"></Column>
+
+                        </DataTablez>
+
+                  </div>
+            </div>
+      )
+}
+
+export default ListUser
Index: src/modules/ListUser/index.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/modules/ListUser/index.js b/src/modules/ListUser/index.js
new file mode 100644
--- /dev/null	
+++ b/src/modules/ListUser/index.js	
@@ -0,0 +1,1 @@
+export { default as ListUser } from './screens/ListUser'
Index: src/axios/index.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/axios/index.js b/src/axios/index.js
new file mode 100644
--- /dev/null	
+++ b/src/axios/index.js	
@@ -0,0 +1,6 @@
+import axios from 'axios';
+
+export const clientApi = axios.create({
+      baseURL: 'http://localhost:8000/',
+      timeout: 10000,
+});
\ No newline at end of file
Index: src/modules/ListUser/utils/index.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/modules/ListUser/utils/index.js b/src/modules/ListUser/utils/index.js
new file mode 100644
--- /dev/null	
+++ b/src/modules/ListUser/utils/index.js	
@@ -0,0 +1,14 @@
+import { useEffect, useState } from "react"
+import { listUser } from "../api"
+
+export const useListUser = (params) => {
+      const [data, setData] = useState([])
+      async function fetchData() {
+            const response = await listUser({ status: 1, ...params })
+            if (response.data.status) setData(response.data.data)
+      }
+      useEffect(() => {
+            fetchData()
+      }, [JSON.stringify(params)])
+      return data
+}
\ No newline at end of file
Index: src/axios/test.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/axios/test.js b/src/axios/test.js
new file mode 100644
--- /dev/null	
+++ b/src/axios/test.js	
@@ -0,0 +1,187 @@
+export const test = [
+
+      {
+            "albumId": 1,
+            "id": 1,
+            "title": "accusamus beatae ad facilis cum similique qui sunt",
+            "url": "https://via.placeholder.com/600/92c952",
+            "thumbnailUrl": "https://via.placeholder.com/150/92c952"
+      },
+      {
+            "albumId": 1,
+            "id": 2,
+            "title": "reprehenderit est deserunt velit ipsam",
+            "url": "https://via.placeholder.com/600/771796",
+            "thumbnailUrl": "https://via.placeholder.com/150/771796"
+      },
+      {
+            "albumId": 1,
+            "id": 3,
+            "title": "officia porro iure quia iusto qui ipsa ut modi",
+            "url": "https://via.placeholder.com/600/24f355",
+            "thumbnailUrl": "https://via.placeholder.com/150/24f355"
+      },
+      {
+            "albumId": 1,
+            "id": 4,
+            "title": "culpa odio esse rerum omnis laboriosam voluptate repudiandae",
+            "url": "https://via.placeholder.com/600/d32776",
+            "thumbnailUrl": "https://via.placeholder.com/150/d32776"
+      },
+      {
+            "albumId": 1,
+            "id": 5,
+            "title": "natus nisi omnis corporis facere molestiae rerum in",
+            "url": "https://via.placeholder.com/600/f66b97",
+            "thumbnailUrl": "https://via.placeholder.com/150/f66b97"
+      },
+      {
+            "albumId": 1,
+            "id": 6,
+            "title": "accusamus ea aliquid et amet sequi nemo",
+            "url": "https://via.placeholder.com/600/56a8c2",
+            "thumbnailUrl": "https://via.placeholder.com/150/56a8c2"
+      },
+      {
+            "albumId": 1,
+            "id": 7,
+            "title": "officia delectus consequatur vero aut veniam explicabo molestias",
+            "url": "https://via.placeholder.com/600/b0f7cc",
+            "thumbnailUrl": "https://via.placeholder.com/150/b0f7cc"
+      },
+      {
+            "albumId": 1,
+            "id": 8,
+            "title": "aut porro officiis laborum odit ea laudantium corporis",
+            "url": "https://via.placeholder.com/600/54176f",
+            "thumbnailUrl": "https://via.placeholder.com/150/54176f"
+      },
+      {
+            "albumId": 1,
+            "id": 9,
+            "title": "qui eius qui autem sed",
+            "url": "https://via.placeholder.com/600/51aa97",
+            "thumbnailUrl": "https://via.placeholder.com/150/51aa97"
+      },
+      {
+            "albumId": 1,
+            "id": 10,
+            "title": "beatae et provident et ut vel",
+            "url": "https://via.placeholder.com/600/810b14",
+            "thumbnailUrl": "https://via.placeholder.com/150/810b14"
+      },
+      {
+            "albumId": 1,
+            "id": 11,
+            "title": "nihil at amet non hic quia qui",
+            "url": "https://via.placeholder.com/600/1ee8a4",
+            "thumbnailUrl": "https://via.placeholder.com/150/1ee8a4"
+      },
+      {
+            "albumId": 1,
+            "id": 12,
+            "title": "mollitia soluta ut rerum eos aliquam consequatur perspiciatis maiores",
+            "url": "https://via.placeholder.com/600/66b7d2",
+            "thumbnailUrl": "https://via.placeholder.com/150/66b7d2"
+      },
+      {
+            "albumId": 1,
+            "id": 13,
+            "title": "repudiandae iusto deleniti rerum",
+            "url": "https://via.placeholder.com/600/197d29",
+            "thumbnailUrl": "https://via.placeholder.com/150/197d29"
+      },
+      {
+            "albumId": 1,
+            "id": 14,
+            "title": "est necessitatibus architecto ut laborum",
+            "url": "https://via.placeholder.com/600/61a65",
+            "thumbnailUrl": "https://via.placeholder.com/150/61a65"
+      },
+      {
+            "albumId": 1,
+            "id": 15,
+            "title": "harum dicta similique quis dolore earum ex qui",
+            "url": "https://via.placeholder.com/600/f9cee5",
+            "thumbnailUrl": "https://via.placeholder.com/150/f9cee5"
+      },
+      {
+            "albumId": 1,
+            "id": 16,
+            "title": "iusto sunt nobis quasi veritatis quas expedita voluptatum deserunt",
+            "url": "https://via.placeholder.com/600/fdf73e",
+            "thumbnailUrl": "https://via.placeholder.com/150/fdf73e"
+      },
+      {
+            "albumId": 1,
+            "id": 17,
+            "title": "natus doloribus necessitatibus ipsa",
+            "url": "https://via.placeholder.com/600/9c184f",
+            "thumbnailUrl": "https://via.placeholder.com/150/9c184f"
+      },
+      {
+            "albumId": 1,
+            "id": 18,
+            "title": "laboriosam odit nam necessitatibus et illum dolores reiciendis",
+            "url": "https://via.placeholder.com/600/1fe46f",
+            "thumbnailUrl": "https://via.placeholder.com/150/1fe46f"
+      },
+      {
+            "albumId": 1,
+            "id": 19,
+            "title": "perferendis nesciunt eveniet et optio a",
+            "url": "https://via.placeholder.com/600/56acb2",
+            "thumbnailUrl": "https://via.placeholder.com/150/56acb2"
+      },
+      {
+            "albumId": 1,
+            "id": 20,
+            "title": "assumenda voluptatem laboriosam enim consequatur veniam placeat reiciendis error",
+            "url": "https://via.placeholder.com/600/8985dc",
+            "thumbnailUrl": "https://via.placeholder.com/150/8985dc"
+      },
+      {
+            "albumId": 1,
+            "id": 21,
+            "title": "ad et natus qui",
+            "url": "https://via.placeholder.com/600/5e12c6",
+            "thumbnailUrl": "https://via.placeholder.com/150/5e12c6"
+      },
+      {
+            "albumId": 1,
+            "id": 22,
+            "title": "et ea illo et sit voluptas animi blanditiis porro",
+            "url": "https://via.placeholder.com/600/45601a",
+            "thumbnailUrl": "https://via.placeholder.com/150/45601a"
+      },
+      {
+            "albumId": 1,
+            "id": 23,
+            "title": "harum velit vero totam",
+            "url": "https://via.placeholder.com/600/e924e6",
+            "thumbnailUrl": "https://via.placeholder.com/150/e924e6"
+      },
+      {
+            "albumId": 1,
+            "id": 24,
+            "title": "beatae officiis ut aut",
+            "url": "https://via.placeholder.com/600/8f209a",
+            "thumbnailUrl": "https://via.placeholder.com/150/8f209a"
+      },
+      {
+            "albumId": 1,
+            "id": 25,
+            "title": "facere non quis fuga fugit vitae",
+            "url": "https://via.placeholder.com/600/5e3a73",
+            "thumbnailUrl": "https://via.placeholder.com/150/5e3a73"
+      },
+      {
+            "albumId": 1,
+            "id": 26,
+            "title": "asperiores nobis voluptate qui",
+            "url": "https://via.placeholder.com/600/474645",
+            "thumbnailUrl": "https://via.placeholder.com/150/474645"
+      },
+
+
+]
\ No newline at end of file
Index: src/redux/store.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/redux/store.js b/src/redux/store.js
new file mode 100644
--- /dev/null	
+++ b/src/redux/store.js	
@@ -0,0 +1,9 @@
+
+import { configureStore } from '@reduxjs/toolkit';
+import userReducer from './userSlice';
+
+export default configureStore({
+      reducer: {
+            user: userReducer,
+      },
+});
Index: src/redux/userSlice.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/redux/userSlice.js b/src/redux/userSlice.js
new file mode 100644
--- /dev/null	
+++ b/src/redux/userSlice.js	
@@ -0,0 +1,31 @@
+// userSlice.js
+import { createSlice } from '@reduxjs/toolkit';
+const tokenFromStorage = localStorage.getItem('token');
+const initialState = {
+      userInfo: null,
+      token: tokenFromStorage || null,
+      isLoggedIn: !!tokenFromStorage,
+};
+
+const userSlice = createSlice({
+      name: 'user',
+      initialState,
+      reducers: {
+            loginSuccess: (state, action) => {
+
+                  state.userInfo = action.payload.userInfo;
+                  state.token = action.payload.token;
+                  state.isLoggedIn = true;
+            }
+            ,
+            logout: (state) => {
+                  state.userInfo = null;
+                  state.token = null;
+                  state.isLoggedIn = false;
+            },
+      },
+});
+
+export const { loginSuccess, logout } = userSlice.actions;
+
+export default userSlice.reducer;
Index: src/components/Card.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/components/Card.js b/src/components/Card.js
new file mode 100644
--- /dev/null	
+++ b/src/components/Card.js	
@@ -0,0 +1,35 @@
+import React from 'react';
+import { Card } from 'primereact/card';
+import { Button } from 'primereact/button';
+import { Avatar } from 'primereact/avatar';
+
+function Cardz({ src, song, sing, link }) {
+
+      const header = (
+            <div class="container">
+                  <img alt="Card" className='h-3 image_overlay' src={src} />
+                  <div class="overlay">
+                  </div>
+                  <div className="play child">
+                        <Avatar className='' icon="pi pi-play" size="xlarge" style={{ color: '#000000' }} />
+                  </div>
+            </div>
+      );
+      const subTitle = () => {
+            return (
+                  <a style={{ color: '#656565', textDecoration: 'none' }} href={link}>{sing}</a>
+            )
+      }
+      return (
+            <div >
+
+                  <Card className='card_hover' title={song} header={header} subTitle={subTitle}>
+                  </Card>
+
+
+
+            </div >
+      )
+}
+
+export default Cardz;
Index: src/components/Datatablez.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/components/Datatablez.js b/src/components/Datatablez.js
new file mode 100644
--- /dev/null	
+++ b/src/components/Datatablez.js	
@@ -0,0 +1,33 @@
+import React from 'react';
+import { DataTable } from 'primereact/datatable';
+import { Column } from 'primereact/column';
+import { Button } from 'primereact/button';
+
+const DataTablez = ({ value, children }) => {
+      const actionBodyTemplate = (rowData) => {
+            return (
+                  <div className='flex gap-2'>
+                        <Button icon="pi pi-file-edit" className=" p-button-success p-mr-2" />
+                        <Button icon="pi pi-trash" className=" p-button-danger" />
+                  </div>
+            );
+      };
+
+      const indexBodyTemplate = (rowData, row) => {
+            return row.rowIndex + 1;
+      };
+
+      const newChildren = [
+            <Column key="stt" body={indexBodyTemplate} header="STT"></Column>,
+            ...React.Children.toArray(children),
+            <Column key="action" body={actionBodyTemplate} header="Hành động"></Column>
+      ];
+
+      return (
+            <DataTable value={value}>
+                  {newChildren}
+            </DataTable>
+      );
+};
+
+export default DataTablez;
Index: src/components/Form.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/components/Form.js b/src/components/Form.js
new file mode 100644
--- /dev/null	
+++ b/src/components/Form.js	
@@ -0,0 +1,24 @@
+import React from 'react';
+import { InputText } from 'primereact/inputtext';
+
+function InputForm(props) {
+      const { value, onChange, placeholder, label, id, ...otherProps } = props;
+      return (
+            <div className="mb-3 px-2 ">
+                  {label && (
+                        <div className="w-full flex justify-content-between">
+                              <label htmlFor={id} className="font-medium w-full pb-2">
+                                    {label}
+                              </label>
+                        </div>
+                  )}
+                  <InputText
+                        id={id}
+                        placeholder={placeholder || (label && `Nhập ${label.toLowerCase()}`)}
+                        value={value}
+                        onChange={onChange} {...otherProps} />
+            </div>
+      );
+}
+
+export default InputForm;
Index: src/components/Title.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/components/Title.js b/src/components/Title.js
new file mode 100644
--- /dev/null	
+++ b/src/components/Title.js	
@@ -0,0 +1,18 @@
+import React from 'react'
+import { Link } from 'react-router-dom'
+
+function Title({ title, SubTitle, subTitle, href, icon }) {
+      return (
+            <div
+                  className=" flex align-items-center mb-2 justify-content-between"
+                  style={{ height: '50px', }}
+            >
+                  <b className="text-3xl p-0">{title}</b>
+                  <Link className='text-xl text-white no-underline pr-2' to={href}>
+                        {subTitle} <i className={`pi ${icon}`} ></i></Link>
+                  {SubTitle && <SubTitle />}
+            </div>
+      )
+}
+
+export default Title
Index: src/layout/AdminLayout/Header/HeaderAdmin.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/layout/AdminLayout/Header/HeaderAdmin.js b/src/layout/AdminLayout/Header/HeaderAdmin.js
new file mode 100644
--- /dev/null	
+++ b/src/layout/AdminLayout/Header/HeaderAdmin.js	
@@ -0,0 +1,17 @@
+import { Button } from 'primereact/button'
+import React from 'react'
+
+function HeaderAdmin() {
+      return (
+            <div className='card h-5rem'>
+                  <div className='flex align-items-center gap-3 justify-content-end px-3'>
+                        <h1 className='text-white text-center' >
+                              Quản trị VietThanhCa
+                        </h1 >
+                        <Button icon='pi pi-sign-out' />
+                  </div>
+            </div>
+      )
+}
+
+export default HeaderAdmin
Index: src/modules/Home/screens/TrendRegular.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/modules/Home/screens/TrendRegular.js b/src/modules/Home/screens/TrendRegular.js
new file mode 100644
--- /dev/null	
+++ b/src/modules/Home/screens/TrendRegular.js	
@@ -0,0 +1,47 @@
+import React from 'react'
+import { test } from '@/axios/test'
+import { Link } from 'react-router-dom'
+function TrendRegular() {
+      return (
+            <div className='trend_regular border-round-lg'>
+                  <ul className='p-0 m-0 flex py-3 justify-content-center'>
+                        <div className='flex  gap-2 flex-column '>
+                              <h3 className='mt-0 px-3'>Thường xuyên nghe</h3>
+                              {
+                                    test.slice(0, 3).map(d =>
+
+                                          <li className='flex align-items-center px-3'>
+                                                <img src="https://picsum.photos/200/300" alt="" width="80px" height="80px" className='border-round-lg' />
+                                                <div className='flex flex-column justify-content-center p-3 gap-2'>
+                                                      <Link to="" className='text-white no-underline text-xl'>Con đường Chúa đã đi</Link>
+                                                      <Link to="" className='text-white no-underline'>Thế Quyền</Link>
+                                                </div>
+                                                {/* <Button icon="pi " /> */}
+                                                <span className="pi pi-ellipsis-v"></span>
+                                          </li>
+                                    )
+                              }
+                              <h3 className='px-3'>Thịnh hành</h3>
+                              {
+                                    test.slice(0, 3).map(d =>
+
+                                          <li className='flex align-items-center px-3'>
+                                                <img src="https://picsum.photos/200/300" alt="" width="80px" height="80px" className='border-round-lg' />
+                                                <div className='flex flex-column justify-content-center p-3 gap-2'>
+                                                      <Link to="" className='text-white no-underline text-xl'>Con đường Chúa đã đi</Link>
+                                                      <Link to="" className='text-white no-underline'>Thế Quyền</Link>
+                                                </div>
+                                                {/* <Button icon="pi " /> */}
+                                                <span className="pi pi-ellipsis-v"></span>
+                                          </li>
+
+                                    )
+                              }
+                        </div>
+
+                  </ul>
+            </div>
+      )
+}
+
+export default TrendRegular
Index: src/modules/Home/index.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/modules/Home/index.js b/src/modules/Home/index.js
new file mode 100644
--- /dev/null	
+++ b/src/modules/Home/index.js	
@@ -0,0 +1,1 @@
+export { default as Home } from './screens/Home'
\ No newline at end of file
Index: src/modules/HomeAdmin/screen/HomeAdmin.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/modules/HomeAdmin/screen/HomeAdmin.js b/src/modules/HomeAdmin/screen/HomeAdmin.js
new file mode 100644
--- /dev/null	
+++ b/src/modules/HomeAdmin/screen/HomeAdmin.js	
@@ -0,0 +1,11 @@
+import React from 'react'
+
+function HomeAdmin() {
+      return (
+            <div>
+                  hi
+            </div>
+      )
+}
+
+export default HomeAdmin
Index: src/modules/HomeAdmin/index.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/modules/HomeAdmin/index.js b/src/modules/HomeAdmin/index.js
new file mode 100644
--- /dev/null	
+++ b/src/modules/HomeAdmin/index.js	
@@ -0,0 +1,1 @@
+export { default as HomeAdmin } from './screen/HomeAdmin'
Index: src/modules/ListSong/screen/ListSong.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/modules/ListSong/screen/ListSong.js b/src/modules/ListSong/screen/ListSong.js
new file mode 100644
--- /dev/null	
+++ b/src/modules/ListSong/screen/ListSong.js	
@@ -0,0 +1,28 @@
+import DataTablez from '@/components/Datatablez'
+import { Column } from 'primereact/column';
+import React from 'react'
+
+
+function ListSong() {
+      const songs = [
+            { id: 1, title: 'Bài hát 1', singer: 'Ca sĩ 1', composer: 'Sáng tác 1' },
+            { id: 2, title: 'Bài hát 2', singer: 'Ca sĩ 2', composer: 'Sáng tác 2' },
+            { id: 3, title: 'Bài hát 3', singer: 'Ca sĩ 3', composer: 'Sáng tác 3' },
+            { id: 4, title: 'Bài hát 4', singer: 'Ca sĩ 4', composer: 'Sáng tác 4' },
+            { id: 5, title: 'Bài hát 5', singer: 'Ca sĩ 5', composer: 'Sáng tác 5' },
+            // Thêm nhiều bài hát khác vào đây
+      ];
+
+      return (
+            <div className='element'>
+                  <DataTablez value={songs}>
+                        <Column field='title' header="Tên bài hát" />
+                        <Column field='singer' header="Ca sĩ" />
+                        <Column field='composer' header="Sáng tác" />
+
+                  </DataTablez>
+            </div>
+      )
+}
+
+export default ListSong
Index: src/modules/ListSong/utils/index.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/modules/ListSong/utils/index.js b/src/modules/ListSong/utils/index.js
new file mode 100644
--- /dev/null	
+++ b/src/modules/ListSong/utils/index.js	
@@ -0,0 +1,11 @@
+export const useListUser = (params) => {
+      const [data, setData] = useState([])
+      async function fetchData() {
+            const response = await listUser({ status: 1, ...params })
+            if (response.data.status) setData(response.data.data)
+      }
+      useEffect(() => {
+            fetchData()
+      }, [JSON.stringify(params)])
+      return data
+}
\ No newline at end of file
Index: src/modules/ListSong/index.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/modules/ListSong/index.js b/src/modules/ListSong/index.js
new file mode 100644
--- /dev/null	
+++ b/src/modules/ListSong/index.js	
@@ -0,0 +1,1 @@
+export { default as ListSong } from './screen/ListSong'
Index: src/modules/ListUser/api/index.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/modules/ListUser/api/index.js b/src/modules/ListUser/api/index.js
new file mode 100644
--- /dev/null	
+++ b/src/modules/ListUser/api/index.js	
@@ -0,0 +1,3 @@
+import { getData } from "@/lib/request";
+
+export const listUser = (params) => getData('user/alluser', params)
diff --git a/src/lib/AuthServices.js b/src/lib/AuthServices.js
new file mode 100644
diff --git a/src/modules/ListSong/api/index.js b/src/modules/ListSong/api/index.js
new file mode 100644
